core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -425,7 +435,10 @@ public Frame decorrelateRel(Sort rel) {
     RelCollation newCollation = RexUtil.apply(mapping, oldCollation);
 
     final Sort newSort =
-        LogicalSort.create(newInput, newCollation, rel.offset, rel.fetch);
+        (Sort) relBuilder
```

Comment:
is it guaranteed that relbuilder would return a sort (for example if newInput has already the correct collation?)

Reply:
I think it is. However, it seems like there is nothing to lose by using RelNode. Changed its type to RelNode.